crux/src/Crux/Goal.hs

```diff
@@ -188,7 +184,7 @@ proveGoalsOffline adapters opts ctx explainFailure (Just gs0) = do
     failfast = proofGoalsFailFast opts
 
     go :: SupportsCruxLogMessage msgs
-       => (Integer -> IO (), IO ())
+       => (Integer -> IO (), Integer -> IO ())
```

Comment:
This type is repeated in a couple of places; would a `type` alias simplify the declarations, or is it still so few places that there's no real benefit?

Reply:
Actually this made me come up with fairly nice type synonyms. Will update!